vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_list_updater.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/publisher/publisher_list_updater.h"
+
+#include <memory>
+#include <utility>
+
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/publisher/publisher_list_reader.h"
+#include "bat/ledger/internal/request/request_publisher.h"
+#include "bat/ledger/internal/state/state_keys.h"
+#include "bat/ledger/option_keys.h"
+#include "brave_base/random.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+
+constexpr double kRetryAfterFailureDelay = 150.0;
+constexpr uint64_t kMaxRetryAfterFailureDelay = 60 * 60 * 4;
+
+}  // namespace
+
+namespace braveledger_publisher {
+
+PublisherListUpdater::PublisherListUpdater(bat_ledger::LedgerImpl* ledger)
```

Comment:
is there a specific reason we can't use the component updater for this? We are trying to standardize on that across all platforms

Reply:
We'll move to component updater in a future PR.